server/location/index.js

```diff
@@ -0,0 +1,175 @@
+const debug = require('debug')('kyf:matcher');
+
+// const { constituencyFromPostcode, locationInfoFromPostcode } = require ('./externals');
+const { isPc7, isPc8, isPostcodeDistrict, isFullPostcode, isPostcode, endsWithPostcode,
+pc7FromFullPostcode, pc8FromFullPostcode, districtFromFullPostcode, districtFromPostcodeDistrict, postcodeFromString,
+toStandardLatLong, toLatLong, isLeafletLatLng, latLongFromString, latLongFrom4dpLatLongString, roundToNearest } = require ('./helpers');
+const { fromPostcodesIo, fromGoogle, fromTwitter,
+  constituencyFromPostcode, locationInfoFromPostcode  } = require ('./requests');
+
+const LocationObject = ()=> ({
+  specificity : 0,
+  twitterString  : '',
+  defaultTwitterFollow : '@uklabour'
+});
+const noVerify = { verify : false };
+
+const cache = {
+  records : 0,
+
+  canonicalise : location => {
+    const latLong = toStandardLatLong(location);
+    if (latLong)
+      return(`${latLong.lng},${latLong.lat}`);                // see note in helpers.js
+    if (isPostcode(location))
+      return pc8FromFullPostcode(location) || districtFromPostcodeDistrict(location)
+    if (endsWithPostcode(location))
+      return postcodeFromString(location)
+    // and then the harder cases, eg London, Hackney, The World
+    return location
+  },
+
+  compress : x=> x,
+
+  uncompress : x=> x,
```

Comment:
what are these for?

Reply:
These are not yet implemented - once implemented they will be to to compress whole Location Objects rather than storing whole Location Objects in the cache